Modules/Algorithms/Shor/shors.ipynb

```diff
@@ -22,68 +22,55 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## 1. Introduction\n",
+    "## 1. Problem Statement\n",
+    "This problem involves factoring a given integer N efficiently. Classical algorithms for integer factorization, such as trial division and the general number field sieve, are slow for large numbers. Shor's Algorithm leverages quantum computing to solve this problem in polynomial time.\n",
     "\n",
-    "### Imports"
+    "Given an integer N, the goal is to find its nontrivial factors using quantum techniques."
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "%pip install numpy\n",
-    "%pip install qiskit\n",
-    "%pip install qiskit_ibm_runtime\n",
-    "%pip install qiskit_aer"
+    "## 2. Shor's Algorithm Steps\n",
+    "\n",
+    "### As an example, say we want to factor $N = 15$."
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 1,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "import numpy as np\n",
-    "import math\n",
-    "from qiskit import QuantumCircuit\n",
-    "from qiskit.circuit.library import CUGate\n",
-    "from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager as gppm\n",
-    "from qiskit_ibm_runtime import QiskitRuntimeService, SamplerV2 as Sampler\n",
-    "from qiskit.visualization import plot_distribution\n",
-    "from qiskit_aer import AerSimulator"
+    "### Classical Preprocessing\n",
+    "#### Step 1. Choose a random integer $a$:\n",
+    "- Pick $a = 2$.\n",
+    "- Find the greatest common divisor.If gcd(a,N)>1, then we already have a factor of N, which means the problem is trivially solved\n",
```

Comment:
Further explain what a greatest common divisor is. Also, missing a space in the middle.

Reply:
Also put the portion about the gcd in TeX. Shouldn't have functions floating around in free text. Applies to all mentions of N.